lib/rucio/core/monitor.py

```diff
@@ -256,12 +257,56 @@ def record_timer(name, time, labels=None):
         timing.observe(time)
 
 
+class Stopwatch:
+    """Stopwatch to measure and record time durations.
+
+    Note: The stopwatch is started on initialization.
+    """
+
+    _t_start: float
+    _t_end: Optional[float]
+
+    def __init__(self) -> None:
+        self.restart()
+
+    def _now(self) -> float:
+        # TODO: change to time.monotonic_ns() if python 3.6 support is dropped.
+        return time.monotonic()
+
+    def restart(self) -> None:
+        """Resets and starts the stopwatch."""
+        self._t_start = self._now()
+        self._t_end = None
+
+    def stop(self) -> None:
+        """Stops the stopwatch."""
+        self._t_end = self._now()
+
+    def record(self, name: str, divisor: float = 1, labels: Optional[Dict] = None) -> None:
+        """Records the currently elapsed time and lets the clock continue running.
+
+        :param name: Name of recorded metric.
+        :param divisor: Optional divisor to scale the elapsed time by.
+        :param labels: Optional dictionary of additional information.
+        """
+        record_timer(name=name, time=self.elapsed / divisor, labels=labels)
+
+    @property
+    def elapsed(self) -> float:
+        """Returns the total number of elapsed seconds."""
+        if self._t_end is None:
+            return self._now() - self._t_start
+        else:
+            return self._t_end - self._t_start
+
+
 class record_timer_block(object):
```

Comment:
I just looked it up and `record_timer_block` seems to never be called with a "divisor". Maybe it's time to simplify it and get rid of the complex divisor logic in it ?

Reply:
It's used with a divisor some places, but never with multiple stats at once, so I'll simplify that.

Reply:
Where is it used with divisor? I only saw a couple of tests and no actual application code using it

Reply:
You were right, the only usages were the ones I introduced in this PR. Removed and simplified now.